msbuild/Xamarin.MacDev.Tasks/Tasks/AOTCompileTaskBase.cs

```diff
@@ -98,10 +100,16 @@ public override bool Execute ()
 					return false;
 				}
 				arguments.Add ($"{string.Join (",", parsedArguments)}");
-				if (globalAotArguments is not null)
+				if (globalAotArguments?.Any () == true)
 					arguments.Add ($"--aot={string.Join (",", globalAotArguments)}");
 				arguments.AddRange (parsedProcessArguments);
-				arguments.Add (input);
+				if (Path.GetFileName (input) == Path.GetFileName (DedupAssembly)) {
+					for (var j = 0; j < Assemblies.Length; j++) {
+						arguments.Add (inputs [j]);
+					}
+				} else {
+					arguments.Add (input);
+				}
```

Comment:
nit: I think introducing an item metadata on the dedup assembly would simplify this.
It would remove the need of having a `DedupAssembly` property and comparing against the filename, where instead we could just check whether the item metadata is set.